xrpl/models/transactions/credential_accept.py

```diff
@@ -0,0 +1,63 @@
+"""Model for CredentialAccept transaction type."""
+
+from dataclasses import dataclass, field
+from typing import Dict, Optional
+
+from typing_extensions import Self
+
+from xrpl.models.required import REQUIRED
+from xrpl.models.transactions.transaction import Transaction
+from xrpl.models.transactions.types import TransactionType
+from xrpl.models.utils import HEX_REGEX, KW_ONLY_DATACLASS, require_kwargs_on_init
+
+
+@require_kwargs_on_init
+@dataclass(frozen=True, **KW_ONLY_DATACLASS)
+class CredentialAccept(Transaction):
+    """This transaction accepts a credential issued to the Account (i.e. the Account is
+    the Subject of the Credential object). The credential is not considered valid until
+    it has been transferred/accepted.
+    """
+
+    transaction_type: TransactionType = field(
+        default=TransactionType.CREDENTIAL_ACCEPT,
+        init=False,
+    )
+
+    account: str = REQUIRED  # type: ignore
+    """
+    The subject of the credential.
+    """
+    issuer: str = REQUIRED  # type: ignore
+    """
+    The issuer of the credential.
+    """
+
+    credential_type: str = REQUIRED  # type: ignore
+    """
+    A (hex-encoded) value to identify the type of credential from the issuer.
+    """
+
+    # Note: Validity checks pertaining to other fields of this transaction require
+    # access to the latest state of the blockchain
+    def _get_errors(self: Self) -> Dict[str, str]:
+        return {
+            key: value
+            for key, value in {
+                **super()._get_errors(),
+                "credential_type": self._get_credential_type_error(),
+            }.items()
+            if value is not None
+        }
```

Comment:
```suggestion
        errors = super()._get_errors()
        if cred_type_error := self._get_credential_type_error() is not None:
            errors["credential_type"] = cred_type_error
        return errors
```

Reply:
I'd like to revert this suggestion -- `self._get_credential_type_error() is not None` only provides a `True/False` value. I want to capture the verbatim value in `self._get_credential_type_error()` and provide that to the user.

Besides, I don't see qualitative difference between the current implementation and the suggestion.

Reply:
`cred_type_error` will always be True/False in the suggested implementation.

Reply:
The code is much more readable this way - you avoid the need of a lambda.

Reply:
To make your suggestion, I'd have to do:
```
errors = super()._get_errors()
        if cred_type_error := self._get_credential_type_error() is not None:
            errors["credential_type"] = self._get_credential_type_error() # second function is not elegant, I'll have to cache it
        return errors
```

cred_type_error is `True`, if there are `credential_type` validity errors.
cred_type_error is `False`, if there are no `credential_type` validity errors.

`self._get_credential_type_error() is not None` is a boolean, that is getting assigned into `cred_type_error`

Reply:
The `:=` operator assigns the variable to `self._get_credential_type_error()`, not to `self._get_credential_type_error() is not None`. 
https://docs.python.org/3/whatsnew/3.8.html#assignment-expressions
I missed some parentheses, is all:
```
        errors = super()._get_errors()
        if (cred_type_error := self._get_credential_type_error()) is not None:
            errors["credential_type"] = cred_type_error
        return errors
```

Reply:
ah okay, I was not familiar with this operator in python. Golang developers use this sort of assignment+conditional check pretty often.

adopted this suggestion in 5425b37e